compiler/rustc_const_eval/src/interpret/machine.rs

```diff
@@ -286,10 +286,12 @@ pub trait Machine<'mir, 'tcx>: Sized {
     ) -> Pointer<Option<Self::PointerTag>>;
 
     /// Convert a pointer with provenance into an allocation-offset pair.
-    fn ptr_get_alloc(
+    /// This reifies a pointer into one with the data expected
+    /// by hooks such as `memory_read` and `memory_written`.
+    fn ptr_reify_alloc(
```

Comment:
Why "reify"? What's wrong with "get"? (Or, `try_get`, due do the `Option`.)
Why does this also return a `Pointer` again?

My guess is that this "resolves" a wildcard provenance to an actual provenance that should be used for a lookup. Is that right? In that case maybe "resolve" or so would be a better term. Also this means we have to be rather careful with the returned point -- it must *only* be used transiently, never stored into the machine state, since we don't want to "fix" its provenance! I wonder if it truly is a good idea to return this as a `Pointer`.